bundles/core/src/main/java/com/adobe/cq/wcm/core/components/sandbox/internal/models/AbstractContainerImpl.java

```diff
@@ -0,0 +1,65 @@
+/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ ~ Copyright 2018 Adobe Systems Incorporated
+ ~
+ ~ Licensed under the Apache License, Version 2.0 (the "License");
+ ~ you may not use this file except in compliance with the License.
+ ~ You may obtain a copy of the License at
+ ~
+ ~     http://www.apache.org/licenses/LICENSE-2.0
+ ~
+ ~ Unless required by applicable law or agreed to in writing, software
+ ~ distributed under the License is distributed on an "AS IS" BASIS,
+ ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ ~ See the License for the specific language governing permissions and
+ ~ limitations under the License.
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/
+package com.adobe.cq.wcm.core.components.sandbox.internal.models;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.models.annotations.injectorspecific.Self;
+import org.apache.sling.models.annotations.injectorspecific.SlingObject;
+
+import com.adobe.cq.wcm.core.components.models.ListItem;
+import com.adobe.cq.wcm.core.components.sandbox.models.Container;
+import com.day.cq.wcm.api.components.Component;
+import com.day.cq.wcm.api.components.ComponentManager;
+
+/**
+ * Abstract class which can be used as base class for {@link Container} implementations.
+ */
+public abstract class AbstractContainerImpl implements Container {
```

Comment:
What are the reasons for having an `abstract` class? We could have a Container model one day and `ContainerImpl` would be its impl.

Reply:
We talked about this offline as well, I actually like the way things are structured currently. I believe we won't have a basic container in the future but rather something smarter, which might include other information (like the current responsive grid); its interface might inherit from `Container` and its implementation from `AbstractContainerImpl`. So it makes sense to keep what we have. We can refactor when we feel it makes sense as it's all implementation details.

Reply:
+1